src/ripple/overlay/impl/PeerImp.h

```diff
@@ -554,11 +553,11 @@ PeerImp::PeerImp (Application& app, std::unique_ptr<beast::asio::ssl_bundle>&& s
     , p_sink_ (app_.journal("Protocol"), makePrefix(id))
     , journal_ (sink_)
     , p_journal_ (p_sink_)
-    , ssl_bundle_(std::move(ssl_bundle))
-    , socket_ (ssl_bundle_->socket)
-    , stream_ (ssl_bundle_->stream)
+    , stream_ptr_(std::move(stream_ptr))
+    , socket_ (stream_ptr_->next_layer())
+    , stream_ (*stream_ptr_)
     , strand_ (socket_.get_executor())
-    , timer_ (beast::create_waitable_timer<waitable_timer>(socket_))
+    , timer_ (waitable_timer{socket_.get_executor()})
```

Comment:
`beast::tcp_stream` has built-in timeouts so if you ever want to get rid of this timer, it can be done.

Reply:
I got rid of several timers. But another ones such as the timer in PeerImp and the timer in websockets stuff control of ping-pong activity. So I don't think that such a timers can be removed.